src/Arcus.Messaging.Pumps.ServiceBus/ServiceBusMessageRouter.cs

```diff
@@ -190,4 +193,168 @@ private async Task AbandonMessageMatchedHandlerFailedAsync(AzureServiceBusMessag
+            {
+                operationName = "Azure Service Bus message processing";
+            }
+
+            context = context is null ? new Dictionary<string, object>() : new Dictionary<string, object>(context);
+            context["ServiceBus-Endpoint"] = serviceBusNamespace;
+            context["ServiceBus-EntityName"] = entityName;
+            context["ServiceBus-EntityType"] = entityType;
+
+            logger.LogWarning("{@Request}", RequestLogEntry.CreateForServiceBus(operationName, isSuccessful, duration, startTime, context));
+        }
+        /// <summary>
+        /// Represents a HTTP request as a log entry.
+        /// </summary>
+        private sealed class RequestLogEntry
+        {
+            private RequestLogEntry(
+                string method,
+                string host,
+                string uri,
+                string operationName,
+                int statusCode,
+                RequestSourceSystem sourceSystem,
+                string requestTime,
+                TimeSpan duration,
+                IDictionary<string, object> context)
+            {
+                RequestMethod = method;
+                RequestHost = host;
+                RequestUri = uri;
+                ResponseStatusCode = statusCode;
+                RequestDuration = duration;
+                OperationName = operationName;
+                SourceSystem = sourceSystem;
+                RequestTime = requestTime;
+                Context = context;
+                Context["TelemetryType"] = "Request";
+            }
+
+            /// <summary>
+            /// Creates an <see cref="RequestLogEntry"/> instance for Azure Service Bus requests.
+            /// </summary>
+            /// <param name="operationName">The name of the operation of the request.</param>
+            /// <param name="isSuccessful">The indication whether or not the Azure Service Bus request was successfully processed.</param>
+            /// <param name="duration">The duration it took to process the Azure Service Bus request.</param>
+            /// <param name="startTime">The time when the request was received.</param>
+            /// <param name="context">The telemetry context that provides more insights on the Azure Service Bus request.</param>
+            /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="duration"/> is a negative time range.</exception>
+            public static RequestLogEntry CreateForServiceBus(
+                string operationName,
+                bool isSuccessful,
+                TimeSpan duration,
+                DateTimeOffset startTime,
+                IDictionary<string, object> context)
+            {
+                return CreateWithoutHttpRequest(RequestSourceSystem.AzureServiceBus, operationName, isSuccessful, duration, startTime, context);
+            }
+
+            private static RequestLogEntry CreateWithoutHttpRequest(
```

Comment:
Isn't it strange to explicitly say 'WithoutHttpRequest' ?  Doesn't look like we have a method which uses a HttpRequest ?
Is this because this piece of code has been copied from Arcus.Observability ?

Reply:
Yes, true, I"ve inlined this method here: https://github.com/arcus-azure/arcus.messaging/pull/643/commits/4253cc5dfdddfd9734189895f7894110004f862f